### PR TITLE
Adding artifact problem test to ensure it has correct dependency path

### PR DESCRIPTION
**Pull request by suztomo**
This is a test that asserts my comment in https://github.com/GoogleCloudPlatform/cloud-opensource-java/pull/1195#pullrequestreview-355381920

### CODE REVIEW

File: dependencies/src/test/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilderTest.java [hunk 1]
```diff
@@ -47,7 +48,9 @@
   @Test
   public void testGetTransitiveDependencies() throws RepositoryException {
     DependencyGraph graph =
-        dependencyGraphBuilder.buildGraph(new Dependency(datastore, "compile")).getDependencyGraph();
```
Comment (suztomo) [inline dependencies/src/test/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilderTest.java]: Line width was exceeding 100 chars.

File: dependencies/src/test/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilderTest.java [hunk 2]
```diff
@@ -60,7 +63,9 @@ public void testGetTransitiveDependencies() throws RepositoryException {
   @Test
   public void testGetCompleteDependencies() throws RepositoryException {
     DependencyGraph graph =
-        dependencyGraphBuilder.buildCompleteGraph(new Dependency(datastore, "compile")).getDependencyGraph();
```
Comment (suztomo) [inline dependencies/src/test/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilderTest.java]: Line width was exceeding 100 chars.

File: dependencies/src/test/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilderTest.java [hunk 3]
```diff
@@ -190,7 +222,8 @@ public void testConfigureAdditionalMavenRepositories_addingGoogleAndroidReposito
     Artifact artifact = new DefaultArtifact("androidx.lifecycle:lifecycle-common-java8:2.0.0");
 
     // This should not raise an exception
-    DependencyGraphResult graph = graphBuilder.buildCompleteGraph(new Dependency(artifact, "compile"));
```
Comment (suztomo) [inline dependencies/src/test/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilderTest.java]: Line width was exceeding 100 chars.

File: dependencies/src/test/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilderTest.java [hunk 4]
```diff
@@ -178,6 +183,33 @@ public void testGetDirectDependencies_respectExclusions() throws RepositoryExcep
...
+        dependencyGraphBuilder.getStaticLinkageCheckDependencyGraph(
+            ImmutableList.of(hibernateCore));
+
+    ImmutableList<UnresolvableArtifactProblem> artifactProblems = result.getArtifactProblems();
+
+    Truth.assertThat(artifactProblems).hasSize(2);
+    UnresolvableArtifactProblem firstProblem = artifactProblems.get(0);
+    assertEquals("xerces:xerces-impl:jar:2.6.2", firstProblem.getArtifact().toString());
+
+    assertEquals(
```
Comment (elharo) [inline dependencies/src/test/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilderTest.java]: can we test this without converting it to a string?

Comment (suztomo) [reply dependencies/src/test/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilderTest.java]: I can setup getter for that. Give me one moment.

Comment (elharo) [reply dependencies/src/test/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilderTest.java]: Maybe not worth it then.

**Comment by suztomo** [comment]
Memo. In case we want not to rely on toString:

```
    // xerces:xerces-impl:jar:2.6.2 was not resolved. Dependency path:
    //   org.hibernate:hibernate-core:jar:3.5.1-Final (compile)
    //     > cglib:cglib:jar:2.2 (compile?)
    //     > ant:ant:jar:1.6.2 (compile?)
    //     > xerces:xerces-impl:jar:2.6.2 (compile?)
    ImmutableList<DependencyNode> dependencyPath = firstProblem.getDependencyPath();

    // org.hibernate:hibernate-core:jar:3.5.1-Final
    // The root node does not need to carry dependency (scope / optional flag)
    assertEquals(Artifacts.toCoordinates(hibernateCore), Artifacts.toCoordinates(dependencyPath.get(0).getArtifact()));

    // cglib:cglib:jar:2.2 (compile?)
    Dependency cglibDependency = dependencyPath.get(1).getDependency();
    assertTrue(cglibDependency.getOptional());
    assertEquals("compile", cglibDependency.getScope());
    assertEquals("cglib:cglib:2.2",
        Artifacts.toCoordinates(cglibDependency.getArtifact()));

    // ant:ant:jar:1.6.2 (compile?)
    Dependency antDependency = dependencyPath.get(2).getDependency();
    assertTrue(antDependency.getOptional());
    assertEquals("compile", antDependency.getScope());
    assertEquals("ant:ant:1.6.2",
        Artifacts.toCoordinates(antDependency.getArtifact()));

    // xerces:xerces-impl:jar:2.6.2 (compile?)
    Dependency xercesDependency = dependencyPath.get(3).getDependency();
    assertTrue(xercesDependency.getOptional());
    assertEquals("compile", xercesDependency.getScope());
    assertEquals("xerces:xerces-impl:2.6.2",
        Artifacts.toCoordinates(xercesDependency.getArtifact()));
```

**Pull request merged**